### ISSUE TITLE
Minimum required properties for competency frameworks

### ISSUE THREAD

**Issue by stuartasutton** [CredentialEngine/vocabularies#469]
The LRMI TG is working on a proposal for schema.org additions to handle describing competency frameworks. Using Phil's proposal phases, like the credential's proposal, the TG is in Phase 1: Preparing for a W3C community launch. While still very much a working document, here's the URL to a google sheet that has the TG's analysis of minimum properties for a proposal:

https://docs.google.com/spreadsheets/d/1Dgr1IthXqE8UtN9qP6GJP8JDHW0OnH2smLkqv-qLi6E/edit#gid=1942234474

Note that there are two paths:

1. CompetencyFramework (subtype of CreativeWork) & Competency (subtype of Intangible); and
2. CompetencyFramework (subtype of DefinedTermSet†) & Competency (subtype of DefinedTerm†)

†DefinedTermSet and DefinedTerm are a new naming of types in schema.org pending as (issue-894) terms:
  
Terms in Pending (Old Names):
>CategoryCode, CategoryCodeSet & codeValue, hasCategoryCode, inCodeSet, CategoryCode

See:
- https://github.com/schemaorg/schemaorg/issues/894
- https://github.com/schemaorg/schemaorg/issues/1711

**Comment by stuartasutton** [comment]
## Proposal:
### COMPETENCY FRAMEWORK (StandardDocument)

Property | Occurrence | Note
-------|-------|-------
author | required (or creatorOf) | literal
creator | required (or author) | object
ctid | required | 
dateCopyrighted | required if Available |
derivedFom |  required if Available |
description | required | 
hasTopChild | recommended |
inLanguage | required |
license | required if available |
name | required | 
publisher | required if available |
publicationStatus | required if available |
source | required if available |
rights | required if available |

### COMPETENCY (Statement)

Property | Occurrence | Note
-------|-------|-------
codedNotation | required if available |
competencyCategory | required if available |
conceptKeyword |recommended | literal
conceptTerm |recommended | object
ctid | required | 
derivedFrom | required if available |
description |required |
educationLevel | required if available |
hasChild | required if available |
inLanguage | required |
isChildOf | required if available |
isPartOf | required | |
license | required if available
listID | required if available |
shouldIndex | recommended | boolean

**Comment by siuc-nate** [comment]
I assume "derivedFrom" would also be required in DerivedCompetencyFramework?

**Comment by stuartasutton** [comment]
Yes, missed it.

**Comment by siuc-nate** [comment]
These have been taken care of:
http://credreg.net/ctdlasn/mapping/registry

**Comment by stuartasutton** [comment]
Follow this in the following google sheet: https://docs.google.com/spreadsheets/d/1E1dXPxLGqNJ5Gr_wxk9sGS6CwuvavhjbaFCXhZmBCf8/edit#gid=1476845803

**Comment by stuartasutton** [comment]
@siuc-nate, I have reopened this issue because the google sheet noted above takes precedent. I'll review to make sure I've not left anything off the google sheet. I did drop `author` and left `creator`. I'm torn as to whether to include `shouldIndex` as being simply too complex to be used outside a context with highly trained personnel...I have not included it in the google sheet for the editor.

**Comment by siuc-nate** [comment]
I see data in the sheet that lists the term, its range, and whether or not it's multi-value, but I don't see anything in regards to which properties are required/recommended/etc.

Has any of that information changed?

**Comment by stuartasutton** [comment]
No, that has not changed. The google sheet identifies what the editor needs to be able to handle. It currently says nothing about required/recommended. 

However, the required/recommended might provide the editing environment with hints of how to group fields for entry etc. I'll look to updating the sheet to include required, required it, recommended.

**Comment by siuc-nate** [comment]
@stuartasutton is this really just a documentation issue?

**Comment by stuartasutton** [comment]
This is a CASS editor problem.

**Comment by stuartasutton** [comment]
I said that [the spreadsheet](http://credreg.net/registry/policy) defined the require/required if available/recommended properties for competencies and competency frameworks. But, we don't have an equivalent competencies page to http://credreg.net/registry/policy. Should this page be extended to include competencies?
